### DISCUSSION TITLE
Use better and more recent KDFs

### DISCUSSION THREAD

**Discussion by MarkusZoppelt** [mprimi/portable-secret#5] [Ideas]
I highly suggest you replace PBKDF2 with Argon2id and SHA1 with SHA256.

PBKDF2 is easier to brute-force and can be accelerated with GPUs. 

SHA-1 is broken and basically deprecated at this point.

**Comment by mprimi** [comment]
I would love to use state-of-the-art KDF, but Argon2 is not part of the [W3C APIs](https://www.w3.org/TR/2017/REC-WebCryptoAPI-20170126/) that [browsers support](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) unfortunately.

So in this case I choose to go with "best encryption I can do in any browser without dependencies" rather than "best encryption available today". It's a conscious tradeoff. 

I have no excuses for SHA-1, I should see about upgrading. 

Thank you bringing this up!
